web-registry/src/lib/ecocredit.ts

```diff
@@ -346,3 +358,133 @@ export const queryEcoClassInfo = async (
     throw new Error(`Error fetching class info: ${err}`);
   }
 };
+
+/**
+ *
+ * REGEN LEDGER API QUERIES
+ *
+ */
+
+/**
+ *
+ * BANK MODULE QUERIES
+ * -------------------
+ *  - AllBalances (TODO)
+ *  - Balance
+ *  - DenomMetadata
+ *  - DenomsMetadata (TODO)
+ *
+ */
+
+// Denom Metadata
+
+export type QueryDenomMetadataProps = {
+  client: BankQueryClient;
+  denom: string;
+};
```

Comment:
I'm not so fond of this pattern that forces you to redefine the request fields separately. This seems like a lot of unnecessary manual work and prone to errors/oversights. We already have those request types generated and available from regen-js: https://github.com/regen-network/regen-js/blob/9e12047790e7779d8a64872528480f0cfd2870ce/packages/api/src/generated/cosmos/bank/v1beta1/query.ts#L1212-L1213

My suggestions:
1. Declare an interface that wraps the `client` that can be extended for defining individual query props:
```
interface BankQueryClientProps {
 client: BankQueryClient;
}
```
2. Use the generated types from regen-js:
```
interface QueryDenomMetadataProps extends BankQueryClientProps {
 request: DeepPartial<QueryDenomMetadataRequest>,
}
```

Another idea would be define take the "extension" approach that is used in cosmjs, see for instance: 
bank extension: https://github.com/cosmos/cosmjs/blob/ae06012a1510ddf48068bbf21374c0bbff3d5bab/packages/stargate/src/modules/bank/queries.ts
how it's being used: https://github.com/cosmos/cosmjs/blob/ae06012a15/packages/stargate/src/stargateclient.ts#L162-L169

Reply:
Yes I like this. The proposal brings a lot of change (structure, pattern), in the details there is still room for improvement of course. In the useQueryLedger hook I was able to define the output a bit better with generic types but the input part is also a WIP..

Reply:
Do you think it's okay if it's something we take into consideration and improve in future iterations/tasks?

Reply:
I'm not sure, I'll need to think through this a bit more
There are still a lot of // TODO in the `useQueryClient` file with lots of `any` type definition which doesn't seem right to merge as is IMO
But I'd welcome other opinions from @haveanicedavid and @mhagel

Reply:
commented above. Agree there are some easy wins here.